runtime/lua/vscode/api.lua

```diff
@@ -493,7 +493,11 @@ function M.with_insert(callback)
     api.nvim_win_set_cursor(0, end_pos)
     startinsert("<Esc>" .. (vim.o.selection ~= "exclusive" and "a" or "i"))
 
-    run_callback(ranges)
+    -- Wait for the cursor synchronization task to complete in VSCode
```

Comment:
I'm not very familiar with this aspect of the codebase, but is there a more robust way to check the status of this task than just waiting for 30ms?

Reply:
I am also aware that this is unreliable. The debounce time is 5ms, so 30ms was chosen. There might be more reliable solutions, but this approach is certainly the simplest.